### PR TITLE
A few small improvements

### PR DESCRIPTION
**Pull request by C4ptainCrunch**
Hi 👋

First of all, thank you @davish for sharing your code 🤗

I made a bunch of small modifications bundled in a single PR (sorry if it's messy).

I wrote a helper function: [`render_turbo()`](https://github.com/hotwire-django/turbo-django/blob/no-js+minimal+helper/turbo/shortcuts.py#L4). That can be used when answering a form submit. If the request comes from trubo, it sends a turbo frame with the correct headers. Otherwise, it it comes from a plain request, it sends a 303 redirect

I also ensured that the demo is still working even when the JS is deactivated by including the form directly on the detail view but also in the /send view.

And some small details:
 * I removed some superfluous tags and attributes: `data-turbo-frame="messages"` in the form as well as an extra `<turbo-frame id="messages-frame">` that did nothing
 * I hacked `sys.path` in the experiment chat app so it can import `turbo` that is in the parent directory (i guess we'll do something prettier when we have a pip installable package)
 * I removed `turbo/migrations/__init__.py` and `turbo/tests.py` because they are empty at the moment

### CODE REVIEW

File: turbo/shortcuts.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    # Either it is a Turbo Frame that initiated the request: we can detect it with
+    # `turbo-stream` in the `Accept` header.
+    if "turbo-stream" in request.META['HTTP_ACCEPT']:
+        # Notify the template that this is a trubo stream response sent over HTTP
+        # (in opposition to turbo steams sent via websockets)
+        context["turbo__is_http"] = True
+
+        # Notify the turbo client that we are sending a turbo-stream message
+        # so the frontend does not try to redirect the user
+        content_type = '%s; turbo-stream;' % content_type
```
Comment (davish) [inline turbo/shortcuts.py:16]:
```suggestion
        content_type = f'{content_type}; turbo-stream;'
```
This is certainly a personal opinion, but I feel like f-strings are a lot more explicit than old-style format strings.

File: experiments/chat/chat/templates/chat/room_detail.html [hunk 2]
```diff
@@ -13,16 +13,15 @@
...
-</turbo-frame>
+<turbo-channels-stream-source model="chat.Message" stream="room_id" value="{{ room.id }}" />
+
+<ul id="messages">
+  {% for message in room.messages.all %}
+      {% include "chat/message.html" with message=message only %}
+  {% endfor %}
+</ul>
+
+{% include "chat/message_form_frame.html" %}
```
Comment (davish) [inline experiments/chat/chat/templates/chat/room_detail.html:24]: This definitely works, and is probably a cleaner to include the form on the back-end based on how the template partials are split up, but it's important to note that including the message form this way means we're not fetching the `send-message` frame on the client -- the old code used the `src` attribute on the frame to fetch the frame from the `/send` page. Either way is fine, and honestly hotwire is so new there probably isn't a "right" way to do this right now, but should probably add a comment here just to make it clear

File: experiments/chat/chat/templates/chat/message_update.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
-{#<turbo-stream action="append" target="message-list">#}
-{#  <template>#}
-{#    {% include 'chat/message.html' with message=message only %}#}
-{#  </template>#}
-{#</turbo-stream>#}
+<turbo-stream action="append" target="messages">
```
Comment (davish) [inline experiments/chat/chat/templates/chat/message_update.html:1]: will there be a duplicate message here on the sender's list? One with `from http` and one with `from WS`?

**Review by davish — APPROVED** [review]
Looks great to me! One small matter of opinion, one note for some explanatory docs, and one question. :D

File: turbo/shortcuts.py (hunk 1 above)
Comment (C4ptainCrunch) [reply turbo/shortcuts.py:16]: It's definitely better and i don't think we want to support python 3.5 so let's use all the modern things :)

File: experiments/chat/chat/templates/chat/message_update.html (hunk 3 above)
Comment (C4ptainCrunch) [reply experiments/chat/chat/templates/chat/message_update.html:1]: Yes indeed, messages will be duplicated. I guess we should have a nice way to choose between getting the turbo-stream from WS or HTTP.
Having an action="replace-or-append" in Turbo Streams would be awesome so a user without WS would still get his own message via HTTP while a user with WS would get messages from every user 🙈

**Pull request closed without merging**